src/test/kotlin/no/nav/syfo/client/OppgaveServiceTest.kt

```diff
@@ -0,0 +1,136 @@
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.time.LocalDate
+
+class OppgaveServiceTest {
+    @MockK
+    private lateinit var oppgaveClient: OppgaveClient
+
+    @MockK
+    private lateinit var metrikk: Metrikk
+
+    private lateinit var oppgaveService: OppgaveService
+
+    @BeforeEach
+    fun setUp() {
+        MockKAnnotations.init(this)
+        oppgaveService = OppgaveService(oppgaveClient, metrikk)
+    }
+
+    @Test
+    fun `skal opprette fordelingsoppgave uten feil`() =
+        runBlocking {
+            val journalpostId = "123"
+            val expectedOppgaveResultat = OppgaveResultat(oppgaveId = 1, duplikat = false, utbetalingBruker = false)
+            coEvery { oppgaveClient.hentOppgaver(any()) } returns OppgaveListeResponse(0, emptyList())
+            coEvery { oppgaveClient.opprettOppgave(any<OpprettOppgaveRequest>()) } returns OpprettOppgaveResponse(1)
+
+            val result = oppgaveService.opprettFordelingsOppgave(journalpostId)
+
+            assertNotNull(result)
+            assertEquals(expectedOppgaveResultat.oppgaveId, result.oppgaveId)
+            assertEquals(expectedOppgaveResultat.utbetalingBruker, result.utbetalingBruker)
+        }
+
+    @Test
+    fun `skal returnere oppgave dersom det finnes fra før`() =
+        runBlocking {
+            val journalpostId = "123"
+            val expectedOppgaveResultat = OppgaveResultat(oppgaveId = 1, duplikat = true, utbetalingBruker = false)
+            coEvery { oppgaveClient.hentOppgaver(any()) } returns
+                OppgaveListeResponse(
+                    1,
+                    listOf(Oppgave(id = 1, oppgavetype = Oppgavetype.INNTEKTSMELDING, prioritet = Prioritet.NORM.name, aktivDato = LocalDate.now())),
+                )
+            val result = oppgaveService.opprettFordelingsOppgave(journalpostId)
+            verify(exactly = 0) {
+                runBlocking { oppgaveClient.opprettOppgave(any<OpprettOppgaveRequest>()) }
+            }
+            assertNotNull(result)
+            assertEquals(expectedOppgaveResultat.oppgaveId, result.oppgaveId)
+            assertEquals(expectedOppgaveResultat.utbetalingBruker, result.utbetalingBruker)
+        }
+
+    @Test
+    fun `skal opprette oppgave uten feil`() =
+        runBlocking {
+            val journalpostId = "123"
+            val aktoerId = "456"
+            val behandlingsKategori = BehandlingsKategori.UTLAND
+            val expectedOppgaveResultat = OppgaveResultat(oppgaveId = 1, false, false)
+            coEvery { oppgaveClient.hentOppgaver(any()) } returns OppgaveListeResponse(0, emptyList())
+            coEvery { oppgaveClient.opprettOppgave(any()) } returns OpprettOppgaveResponse(1)
+            coEvery { metrikk.tellOpprettOppgave(any()) } just Runs
+
+            val result = oppgaveService.opprettOppgave(journalpostId, aktoerId, behandlingsKategori)
```

Comment:
tenker det er greit å teste alle ulike casene for opprettelse av oppgave også. 
så her kan man feks gjøre 
`           coVerify {
                oppgaveClient.opprettOppgave(match { it.journalpostId == journalpostId && it.aktoerId == aktoerId && it.behandlingstype == "ae0106"  && it.behandlingstema == null })
            }`